src/Symfony/Component/HttpClient/UriTemplateHttpClient.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class UriTemplateHttpClient implements HttpClientInterface, ResetInterface
+{
+    use DecoratorTrait;
+
+    /**
+     * @var  \Closure(string $url, array $vars): string
+     */
+    private \Closure $expander;
+
+    /**
+     * @param (\Closure(string $url, array $vars): string)|null $expander
+     */
+    public function __construct(
+        HttpClientInterface $client = null,
+        \Closure $expander = null,
+    ) {
+        $this->client = $client ?? HttpClient::create();
+        $this->expander = $expander ?? $this->createExpanderFromPopularVendors();
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        $vars = $options['vars'] ?? null;
```

Comment:
this is missing support for vars defined via default options (we need to redefine withOptions for this purpose also)